trie/hexary.py

```diff
@@ -131,20 +153,88 @@ def _get(self, root_hash, trie_key):
         else:
             raise Exception("Invariant: This shouldn't ever happen")
 
-    def _traverse(self, root_hash, trie_key) -> Tuple['Node', Tuple[int, ...]]:
+    def traverse(self, trie_key_input: Nibbles) -> TrieNode:
+        """
+        Find the node at the path of nibbles provided. Unspecified behavior if
+        path ends in the middle of an extension.
+
+        :raises MissingTraversalNode: if a node body is missing from the database
```

Comment:
As well as if the path ends in an extension node?

Reply:
That's the `TraversedPartialPath` one line below